lib/cli-api/_common/_cleanSiteIdForConnectedApp.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+// Initialize constants
+const config = require('config');
+
+/**
+ * @function _cleanSiteIdForConnectedApp
+ * @description Helper function to take a siteId and clean-up so that it can be included
+ * in the corresponding salesforcePlatform connectedApp ID name.
+ *
+ * @param siteId {String} Represents the siteId that is being incorporated into a connectedApp name
+ * @returns {String} Returns the cleaned-up siteId that is now safe to use in the connectedApp name
+ */
+module.exports = function _cleanSiteIdForConnectedApp(siteId) {
+
+    // Initialize local variables
+    let output,
+        lastCharacter,
+        firstCharacter;
+
+    // First, remove whitespace
+    output = siteId.trim();
+
+    // Next, replace all whitespace found in the site definition
+    output = output.replace(config.get('connectedAppIdRegEx.whiteSpace'), '');
+
+    // Next, replace all dashes with an underscore (to try and maintain some consistency)
+    output = output.replace(config.get('connectedAppIdRegEx.dashes'), '_');
+
+    // Replace double-underscores until none exist
+    output = output.replace(config.get('connectedAppIdRegEx.doubleUnderscore'), '_');
+
+    // Next, remove all non-alpha-numeric non-underscore characters
+    output = output.replace(config.get('connectedAppIdRegEx.nonAlphaNumeric'), '');
```

Comment:
As you are doing all the same stuff each time, you could refactor this to loop through the configuration.
Let's move the replacement character into the configuration first:
```javascript
// Define the regular expressions used to govern the siteId / connectedAppId
"connectedAppIdRegEx": {
    // Describe the individual regExes used to clean the siteId
    "whiteSpace": {
    	"regex": /\s/g,
    	"replacement": ""
    },
    "dashes": {
    	"regex": /-/g,
    	"replacement": "_"
    },
    "doubleUnderscore": {
    	"regex": /_+/g,
    	"replacement": "_"
    },
    "nonAlphaNumeric": {
    	"regex": /[^\w_]+/g,
    	"replacement": ""
    }
},
"connectedAppIdFirstCharacterRegEx": /[a-zA-Z]/g
```

```suggestion
const regexs = config.get('connectedAppIdRegEx');
output = Object.keys(regexs).reduce((siteId, regexConfig) => {
    return siteId.replace(regexConfig.regex, regexConfig.replacement
}, output);
```

Reply:
Nice!  Will do -- thank you! :)